contracts/staking/src/lib.rs

```diff
@@ -306,54 +304,50 @@ extern fn init() {
     unsafe { STAKING = Some(staking) };
 }
 
-fn common_state() -> IoStaking {
-    let state = static_mut_state();
-
-    let Staking {
-        owner,
-        staking_token_address,
-        reward_token_address,
-        tokens_per_stake,
-        total_staked,
-        distribution_time,
-        produced_time,
-        reward_total,
-        all_produced,
-        reward_produced,
-        stakers,
-        transactions,
-        current_tid,
-    } = state.clone();
-
-    let stakers = stakers.iter().map(|(k, v)| (*k, v.clone())).collect();
-
-    IoStaking {
-        owner,
-        staking_token_address,
-        reward_token_address,
-        tokens_per_stake,
-        total_staked,
-        distribution_time,
-        produced_time,
-        reward_total,
-        all_produced,
-        reward_produced,
-        stakers,
-        transactions,
-        current_tid,
-    }
-}
-
-fn static_mut_state() -> &'static mut Staking {
-    unsafe { STAKING.get_or_insert(Default::default()) }
-}
-
 #[no_mangle]
 extern fn state() {
-    reply(common_state())
-        .expect("Failed to encode or reply with `<AppMetadata as Metadata>::State` from `state()`");
+    let staking = unsafe { STAKING.as_ref().expect("Unexpected error in taking state") };
+    msg::reply::<IoStaking>(staking.into(), 0)
+        .expect("Failed to encode or reply with `IoStaking` from `state()`");
 }
 
-fn reply(payload: impl Encode) -> GstdResult<MessageId> {
-    msg::reply(payload, 0)
+impl From<&Staking> for IoStaking {
```

Comment:
And here

Reply:
OK, I will change this point in all contracts